### ISSUE TITLE
Old style callback is deprecated (XGBoost 1.3.1)

### ISSUE THREAD

**Issue by GuidoBartoli** [optuna/optuna#2146]
While using the `optuna.integration` module to activate pruning via callback, the latest version of XGBoost gives a warning about the deprecation of the supplied callback style.

## Expected behavior

Despite displaying this message, the pruning procedure seems to be enabled anyway, but since it is deprecated I think it is not guaranteed to work in the future.

## Environment

- Optuna version: **2.3.0**
- Python version: **3.8.5**
- OS: **Linux Mint 20** (based on Ubuntu 20.04 with kernel 5.4.0)
- Other libraries and their versions: **XGBoost 1.3.1**

## Error messages, stack traces, or logs

```
UserWarning: Old style callback is deprecated.  See: https://xgboost.readthedocs.io/en/latest/python/callbacks.html
```

## Steps to reproduce

- Run the code of the official [example](https://github.com/optuna/optuna/blob/master/examples/pruning/xgboost_integration.py)

## Reproducible examples

```python xgboost_integration.py```

**Comment by toshihikoyanase** [comment]
Thank you for your detailed report.
I think it was resolved by #2078, so please use the master branch or wait for the release of Optuna v2.4.0.

**Comment by GuidoBartoli** [comment]
Ok, I will wait for the next release, thanks!
